appservice/src/createAppService/createAppService.ts

```diff
@@ -35,6 +37,16 @@ export async function createAppService(
         environment: node.environment
     };
 
+    if (resourceGroup) {
```

Comment:
Feels like you should just set `wizardContext.newResourceGroupName = resourceGroup` like you do [here](https://github.com/Microsoft/vscode-azuretools/blob/master/appservice/src/createAppService/createAppService.ts#L96)

That would be simpler, plus I think it makes sense to create the resource group if it doesn't exist

Reply:
Yeah, I suppose it makes sense to just create the rg for them then.

Reply:
So that logic relies on the `advancedCreation` being `false`.

The step is being added here https://github.com/Microsoft/vscode-azuretools/blob/master/appservice/src/createAppService/createAppService.ts#L78.  This is neccessary to execute the rg create step which will find/create the rg if neccessary.  In order to make it work for Functions, we'd have to add some more logic.  

Thinking it might be better to just go back to the find logic, and if it can't find it, it'll be undefined and the user will just be prompted for an rg name.

Reply:
So my thoughts are I'll go ahead and merge this because:
1. The use case seems to be for a predetermined rg, so creating it for them seems like a nice to have.
1. This way touches less code and the consequence for incorrectly writing the rg name is to be prompted rather than creating a new rg with the wrong name.
1. If we were to add the creation, we'd have to add some logic to `ResourceListStep()` to create if it only detects a `newResourceGroupName` which could have potential regressions/unexpected side effects.

If we don't want this, let me know and I can fix it up before we release.

Reply:
I think creating it for them is more than just a "nice to have". I think the scenario is less about using an existing resource group and more about putting a bunch of new resources in the same resource group. So depending on which order they create those resources in, it's entirely possible the rg won't exist yet

I'm intimately familiar with the wizard code haha and I'm confident you can just add `executeSteps.push(new ResourceGroupCreateStep());` to the function app case you referenced above and it will work. I definitely think the risk is worth the value considering CTI hasn't started testing yet